test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedDeserializerUndeclaredTests.cs

```diff
@@ -1748,6 +1750,237 @@ public void ReadNonOpenEntryEdmUntypedPropertyInComplexTest()
         }
         #endregion
 
+        #region declared Edm.Untyped property with odata.type
+        [Fact]
+        public void ReadSingleValueEdmUntypedPropertyWithODataTypeResourceTest()
+        {
+            const string payload = @"{
+  ""MyEdmUntypedProp1"":{
+    ""@odata.type"": ""#Server.NS.Address"",
+    ""Street"":""Mars Rd""
+  }
+}";
+            ODataResource topLevelResource = null;
+            ODataResource nestedResource = null;
+            ODataNestedResourceInfo nestedInfo = null;
+            this.ReadEntryPayload(payload, this.serverEntitySet, this.serverEntityType, reader =>
+            {
+                if (reader.State == ODataReaderState.ResourceStart)

```

Comment:
Would using a `switch` statement be better?

Reply:
Yes, a switch could be better. 
But in this test class, can i leave it so I can follow up with the existing test case pattern?